core/deployment/src/main/java/io/quarkus/deployment/configuration/FloatConfigType.java

```diff
@@ -54,8 +56,8 @@ public void generateAcceptConfigurationValue(final BytecodeCreator body, final R
     public void acceptConfigurationValueIntoGroup(final Object enclosing, final Field field, final NameIterator name,
             final SmallRyeConfig config) {
         try {
-            final Float floatValue = config.getValue(name.toString(), Float.class);
-            final float f = floatValue != null ? floatValue.floatValue() : 0f;
+            final Object floatValue = ConfigUtils.getValue(config, name.toString(), Float.class, converterClass);
```

Comment:
Another point where making `ConfigUtils.getValue` be generic will clean up the code.